src/main/java/com/streisky/virtualshowcasebackend/controller/product/ProductController.java

```diff
@@ -0,0 +1,42 @@
+package com.streisky.virtualshowcasebackend.controller.product;
+
+import com.streisky.virtualshowcasebackend.dto.product.ProductDTO;
+import com.streisky.virtualshowcasebackend.service.product.ProductService;
+import jakarta.transaction.Transactional;
+import jakarta.validation.Valid;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.web.PageableDefault;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping("product")
+public class ProductController {
+
+    @Autowired
+    private ProductService productService;
+
+    @PostMapping
+    @Transactional
+    public void register(@RequestBody @Valid ProductDTO productDTO) {
+        productService.save(productDTO);
+    }
+
+    @PutMapping
+    @Transactional
+    public void update(@RequestBody @Valid ProductDTO productDTO) {
+        productService.save(productDTO);
+    }
+
+    @DeleteMapping("/{id}")
+    @Transactional
+    public void delete(@PathVariable Long id) {
+        productService.delete(id);
```

Comment:
I think your probably gonna do this on a separate PR, but it would be nice to have error handling, such as returning `404 Not Found` if a invalid id is received.

Reply:
Created the VirtualShowcaseNotFoundException and handled it to return the corresponding HTTP Status on the endpoint.